modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java

```diff
@@ -268,7 +294,7 @@ public IborCapletFloorletPeriodAmounts impliedVolatilities(
         .filterKeys(period -> volatilities.relativeTime(period.getFixingDateTime()) >= 0)
```

Comment:
maybe use rates provider here to check for expiration to keep consistent with forward rates

Reply:
The filter here is reusing the same logic contained in the original implied vol calculation in the period pricer, and purely used to stop exceptions being thrown by the period pricer's ArgChecker when calling in the stream.

Reply:
Additionally, the `validate()` method checks that the rates provider and volatilities have the same valuation date, so it wouldn't offer any functionality.

Reply:
In strata a client could potentially pass an IborCapletFloorletVolatilities that is time dependent, so it can handle intraday. In this case volatilities.relativeTime(period.getFixingDateTime()) >= 0) might not equate to ratesProvider.getValuationDate().isAfter(period.getFixingDate()). This means we could have a case where implied volatility assumes trade hasn't expired where as for forward rates trade is presumed expired

Reply:
The check using the valuation date from the rates provider `forwardRates()` method is basically if the rates provider date is before or on - if a time were available then we'd check that too but the rates provider doesn't provide. If I check using the logic from the `forwardRates()` method instead of checking against the volatilities date and time, then if the volatilities are intraday, the method could throw an exception in the period pricer if we are on the fixing date but after the vol expiry.